WMFData/Sources/WMFData/Data Controllers/Activity Tab/WMFActivityTabDataController.swift

```diff
@@ -185,6 +188,177 @@ public final class WMFActivityTabDataController {
         guard let project = WMFProject(id: page.projectID) else { return nil }
         return try await articleSummaryController.fetchArticleSummary(project: project, title: page.title)
     }
+
+    // MARK: - Experiment
+
+    @objc public func assignmentIntBlocking() -> Int {
+        let semaphore = DispatchSemaphore(value: 0)
+        var result = 0
+
+        Task.detached { [weak self] in
+            defer { semaphore.signal() }
+            guard let self else { return }
+
+            do {
+                let assignment = try await self.assignOrFetchExperimentAssignment()
+                if assignment == .activityTab {
+                    result = 1
+                } else {
+                    result = 0
+                }
+            } catch {
+                result = 0
+            }
+        }
+
+        semaphore.wait()
+        return result
+    }
+
+    public func assignOrFetchExperimentAssignment() async throws -> ActivityTabExperimentAssignment? {
```

Comment:
We typically need to post the analytics group assignment at the moment of assignment. So it may be helpful to split these bits of logic up, so that upon launch somewhere code could look something like:

```
// Somewhere upon launch
if dataController.needsGroupAssignment() { //this method can check dev settings flags, start/end dates, if assignment is null
     let assignment = dataController.assignGroup()
     ActivityTabFunnel.shared.logAssignment(assignment)
} 

if dataController.needsActivityTab { //this method can dev settings force flags, assignment
    //embed Activity into 4th tab
}
```